### PR TITLE
Add external endpoint validation

### PR DESCRIPTION
**Pull request by SavinduDimal**
Fix wso2/api-manager/issues/148

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/PublisherCommonUtils.java [hunk 1]
```diff
@@ -938,6 +942,33 @@ private static void extractURLsFromEndpointConfig(org.json.JSONObject endpointCo
...
+     * Extract sandbox and production external endpoint URLs and external dev portal URL.
+     *
+     * @param apiDto        API DTO of the API
+     * @param endpoints     List of URLs. Extracted URL(s), if any, are added to this list.
+     */
+    private static void extractExternalEndpoints(APIDTO apiDto, ArrayList<String> endpoints) {
+
+        if (apiDto != null && apiDto.getAdvertiseInfo() != null) {
+            AdvertiseInfoDTO advertiseInfoDto = apiDto.getAdvertiseInfo();
+            if (Boolean.TRUE.equals(advertiseInfoDto.isAdvertised())) {
```
Comment (dushaniw) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/PublisherCommonUtils.java:955]: shall we add a unit test cases for this (covering dto, endpoint null scenarios as well as url validation test cases)?

**Review by dushaniw — APPROVED** [review]
